### DISCUSSION TITLE
Communication between apps and sites

### DISCUSSION THREAD

**Discussion by DeeJaayArgh** [h2oai/wave#1802] [Q&A] [answered]
Is there any built-in capability in h2o-wave to allow some control over a site from an app, particularly if the app was displaying the site in a ui.frame.   As an example, take the dashboard site provided in examples, would there be a way to hide some dashboard items from a separate h2owave app?

I could imagine doing this via some messaging protocol or database connection, but was just curious if there was a solution within the design of h2o-wave.

**Comment by mturoci** [comment, marked as answer]
Interesting question. This should be possible directly from Wave itself:

```py
from h2o_wave import Q, ui, main, app


@app('/')
async def serve(q: Q):
    if not q.client.initialized:  # First visit, create an empty form card for our wizard
        q.page['wizard'] = ui.form_card(box='1 1 2 4', items=[
            ui.text(content='Initially, /demo should be empty.'),
            ui.buttons([ui.button(name='update_site', label='Update site', primary=True)]),
        ])
        q.client.initialized = True

    if q.args.update_site:
        # Get existing or create new site.
        page = q.site['/demo']

        # Add some content.
        page['hello'] = ui.markdown_card(
            box='1 1 2 2',
            title='Hello World!',
            content='Changed',
        )

        # Finally, sync the page to send our changes to the server.
        await page.save()
        q.page['wizard'].items[0].text.content = 'Now, /demo should have a card.'

    await q.page.save()
```

Not sure if the iframe would reload automatically or you would need to update it manually.

Edit: This assumes both app and your static site (`q.site['/demo']`) are running against the same Wave server, which is 99% the case anyway.

**Reply by DeeJaayArgh** [reply]
Thanks!

**Reply by mturoci** [reply]
Please mark the answer as accepted if it answered your question - helps other people when looking for the same.
